js/vlanlist.js

```diff
@@ -0,0 +1,16 @@
+﻿// $Id$
+
+function savecolumnstate() {
+    var colchecked = [];
+    $.each($("input[class='columns']:checked"), function() {
+        colchecked.push(this.id)
+    });
+
+    var jsonString = JSON.stringify(colchecked);
+
+    $.ajax({
+        url: '?m=vlanlist',
+        type: "POST",
+        data: { visiblecolumns : jsonString },
+    });
+}
```

Comment:
Czy rzeczywiście warto zapisywać na serwerze widoczne kolumny? Może lepiej wykorzystać do tego localStorage lub sessionStorage w przeglądarce www?

Reply:
Zastanawiałem się nad tym i uważam, że tak ponieważ to ustawienie w większości konfigurujemy raz dla roli a korzystamy na wielu urządzeniach.

Reply:
To w takim razie sugeruję  wysyłać z content-type apllication/json i nie robić ręcznego JSON.stringify().

Reply:
No i uważam, że lepiej byłoby jakiś ogólny mechanizm w CORE zaprojektować w CSS/JS, który by pozwalał łatwo przenosić widoczność kolumn do różnych formularzy.

Reply:
> To w takim razie sugeruję wysyłać z content-type apllication/json i nie robić ręcznego JSON.stringify().

Dorobiłem.

> No i uważam, że lepiej byłoby jakiś ogólny mechanizm w CORE zaprojektować w CSS/JS, który by pozwalał łatwo przenosić widoczność kolumn do różnych formularzy.

Ja z CSSów jestem jeszcze na poziomie podstawówki. 
Jeśli masz pomysł jak to udoskonalić i przenieść do CORE zapraszam do kontrybucji.

Reply:
Tu dużo CSS nie jest potrzebne - wystarczy, gdybyśmy kolumny z opcją włączania/wyłączania widoczności oznaczali jedynie w `<THEAD>` jakąś umowną klasą np. `lms-ui-table-column-visibility-toggle` plus przez atrybut `id`, żeby zapewnić info o nazwach kolumn i potem kawałkiem kodu w JS robili resztę.

Reply:
no i jeszcze w `<table>` można by używać jego atrybutu `id` jako identyfikatora tabeli, żeby wiedzieć na serwerze jakiej tabeli dotyczy przesłanie stanu widoczności kolumn.

Reply:
No i do wyboru widocznych kolumn można by użyć `lms-ui-multiselect` w trybie `tiny`.

Reply:
> no i jeszcze w <table> można by używać jego atrybutu id jako identyfikatora tabeli, żeby wiedzieć na serwerze jakiej tabeli dotyczy przesłanie stanu widoczności kolumn.

@chilek puść proszę commit jak to widzisz, ja w pierwszej kolejności próbowałem tak - gdzieś miałem mur dlatego zrobiłem tak jak mi się udało, faktycznie zminimalizowałoby to kod troszeczkę.

Reply:
> No i do wyboru widocznych kolumn można by użyć lms-ui-multiselect w trybie tiny.

Wiesz co? Myślałem też o tym. 
Uważam, że widoczność filtrów, kryterium sortowania oraz widoczność kolumn możemy włożyć pod przycisk, żeby nie śmiecić w UI. Tego nie zmienia się często - a po ustawieniu odpowiednim filtrów stałych praktycznie z filtrów się z tych zmian nie korzysta.

Reply:
> z ręcznych zmian filtrów się nie korzysta. I uważam, że do tego należy dążyć - zrobiłem już kiedyś research w tym kierunku i tak jest w kilku produktach.

No właśnie, a przy tym lms-ui-multiselect jest od jakiegoś czasu responsywny.

Reply:
No to może podejść do tematu inaczej:
- dorzucić filtry stałe,
- sprawić, żeby zapamiętywały również widoczność kolumn,
- wybór filtra nie zapamiętywać per login tylko per sesja
?